algorithms/src/sorting/impl/Kokkos_SortByKeyImpl.hpp

```diff
@@ -87,7 +87,19 @@ static_assert_is_admissible_to_kokkos_sort_by_key(const ViewType& /* view */) {
                 "LayoutRight, LayoutLeft or LayoutStride.");
 }
 
+// For the fallback sort_by_key implementation we might need to execute
+// Kokkos::sort on the host. This type trait determines at compile-time where we
+// want to execute.
+template <class ExecutionSpace, class Layout>
+struct sort_on_device : std::false_type {};
```

Comment:
Why bother with the trait?  It doesn't seem like you use it.  You can directly specialize the variable template.